tools/ruby-gems/udb/lib/udb/cli/create.rb

```diff
@@ -0,0 +1,689 @@
+            - where variable _operand_ fields occur (e.g., rd in bits 11:7).
+
+          An instruction _sub type_ is a child of an instruction _type_ that attaches semantic
+          information to the variable operands (e.g., rd is an X destination register).
+
+          There are few instruction types (e.g., I, R, U, B, S)
+          but many instruction sub types (e.g., R-x, R-f, R-x-i)
+
+          Press any key to continue
+        INFO
+
+        TtyTools.display_box(prompt, "Subtypes", type_subtype_info)
+
+        subtype = prompt.select \
+          "What is the subtype of the instruction format?",
+          subtypes,
+          filter: true
+
+        type_contents = nil
+        subtype_contents = nil
+        if subtype == "Create new"
+          files.concat CreationActions.create_inst_subtype(prompt, copyright, outdir)
+          subtype_file = T.must(files.find { |f| f.path.to_s =~ /inst_subtype/ })
+          subtype_contents = YAML.load(subtype_file.contents)
+          type_ref = subtype_contents.fetch("data").fetch("type").fetch("$ref")
+          $stderr.puts type_ref
+          type_paths = Dir[Udb.repo_root / "spec" / "**" / type_ref.gsub("#", "")]
+          if type_paths.empty?
+            type_file = T.must(files.find { |f| f.path.to_s =~ /inst_type/ })
+            type_contents = YAML.load(type_file.contents)
+          else
+            type_contents = YAML.load_file(type_paths.fetch(0))
+          end
+        else
+          subtype_paths = Dir[Udb.repo_root / "spec" / "std" / "isa" / "inst_subtype" / "*" / "#{subtype}.yaml"]
+          raise "can't find subtype" unless subtype_paths.size == 1
+          subtype_contents = YAML.load_file(subtype_paths.fetch(0))
+          type_ref = subtype_contents.fetch("data").fetch("type").fetch("$ref")
+          type_paths = Dir[Udb.repo_root / "spec" / "std" / "isa" / type_ref.gsub("#", "")]
+          if type_paths.empty?
+            raise "Can't find type path"
+          else
+            type_contents = YAML.load_file(type_paths.fetch(0))
+          end
+        end
+        type_name = type_contents["name"]
+        subtype_name = subtype_contents["name"]
+
+        opcode_info = <<~INFO
+          The #{subtype_name} subtype has #{type_contents.fetch("opcodes").size} opcodes and #{subtype_contents.fetch("data").fetch("variables").size} variables.
+
+          Everything needed for variables is provided by the subtype.
+
+          For opcodes, you will provide a _display name_ for documentation (such as encoding diagrams)
+          and a specific _value_ for the instruction.
+
+          We distinguish between two classes of opcodes:
+            - Those that are shared by multiple instructions (e.g., OP, OP-32 in encoding[6:0])
+            - Those that are specific to an instruction (e.g., ADDI in encoding[14:12])
+
+          Common opcodes become their own object in the database and referenced by instructions.
```

Comment:
```suggestion
          Shared opcodes become their own object in the database and referenced by instructions.
```
...to match the preceding language.